ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java

```diff
@@ -530,8 +555,15 @@ private void setGenesis(final Block genesisBlock) {
         throw new IllegalStateException("Blockchain is missing genesis block data.");
       }
       if (!genesisHash.get().equals(genesisBlock.getHash())) {
+        final String firstLine;
+        if (dataDirectory != null) {
+          firstLine =
+              "Supplied genesis block does not match stored chain data in " + dataDirectory + "\n";
```

Comment:
might be cleaner to make firstLine a StringBuffer and append to it, or make that first bit a const

Reply:
1368608